lib/rules/jsx-indent.js

```diff
@@ -165,8 +184,9 @@ module.exports = {
       } while (token.type === 'JSXText' && /^\s*$/.test(token.value));
       var startLine = node.loc.start.line;
       var endLine = token ? token.loc.end.line : -1;
+      var whitespaceOnly = token ? /\n\s*$/.test(token.value) : false;
```

Comment:
I just tried out changing it to `^` and it results in an additional extraneous error in one of the test cases. Specifically:

```javascript
function MyComponent(props) {
	return (
    <div
			className="foo-bar"
			id="thing"
    >
      Hello world!
    </div>
	)
}
```

Because `token.value` is actually everything between the `>` and the `</div>` (which includes a `\n`)